streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -172,7 +172,7 @@
                                         REPLICATION_FACTOR_DOC)
                                 .define(TIMESTAMP_EXTRACTOR_CLASS_CONFIG,
                                         Type.CLASS,
-                                        ConsumerRecordTimestampExtractor.class.getName(),
+                                        FailingConsumerRecordTimestampExtractor.class.getName(),
```

Comment:
Because it is the default extractor, use will not explicitly set it. Thus it should not be a big issue.
Of course, strictly it is a breaking change and we must mention it in the KIP (but we need a KIP anyway).